### PR TITLE
Add unit conversion method to UnitResolver

### PR DESCRIPTION
**Pull request by connorcoley**
Addresses #693

### CODE REVIEW

File: ord_schema/units.py [hunk 1]
```diff
@@ -138,6 +138,63 @@
     "m": "ambiguous between meter and minute",
 }
 
+_UNIT_CONVERSIONS = {
```
Comment (connorcoley) [inline ord_schema/units.py:141]: Since I didn't add an exhaustive set of test cases, would appreciate another pair of eyes on this (@skearnes, @qai222)

**Review by FanwangM — APPROVED** [review]
The unit conversions look good to me.

File: ord_schema/units.py [hunk 2]
```diff
@@ -237,17 +287,109 @@ def resolve(self, string: str, allow_range: bool = False) -> ord_schema.UnitMess
...
+            message: a message with units, e.g., Mass, Length.
+            new_units: the desired units of the new message, expressed
+                either as a string or an integer (ENUM value). Use of a
+                string is recommended due to the ambiguity of using
+                ENUM values; for example, Mass.GRAM == Time.MINUTE.
+
+        Returns:
+            A new message with units, e.g., Mass, Length.
+        """
+        #pylint: disable=too-many-branches
```
Comment (qai222) [inline ord_schema/units.py:331]: looks like the missing space after `#` is causing errors in format tests (GitHub actions)

File: ord_schema/units.py [hunk 3]
```diff
@@ -138,6 +138,63 @@
...
+    },
+    reaction_pb2.Current: {
+        reaction_pb2.Current.AMPERE: 1,
+        reaction_pb2.Current.MILLIAMPERE: 0.001,
+    },
+    reaction_pb2.Voltage: {
+        reaction_pb2.Voltage.VOLT: 1,
+        reaction_pb2.Voltage.MILLIVOLT: 0.001,
+    },
+    reaction_pb2.FlowRate: {
```
Comment (qai222) [inline ord_schema/units.py:189]: Am I crazy or `1 uL/min` should be `1/60 uL/s`?

Comment (connorcoley) [reply ord_schema/units.py:189]: My strategy here was to peg everything to a reference of 1 'unit'. So for example, one `VOLT` is 1 unit (V), one `MILLIVOLT` is 0.001 unit (V).

One `MICROLITER_PER_SECOND` is 60x as high of a flowrate as one `MICROLITER_PER_MINUTE`

File: ord_schema/units.py (hunk 2 above)
Comment (connorcoley) [reply ord_schema/units.py:331]: Thanks for the catch

**Review by qai222 — APPROVED** [review]

File: ord_schema/units.py [hunk 4]
```diff
@@ -294,7 +294,7 @@ def resolve(self, string: str, allow_range: bool = False) -> ord_schema.UnitMess
             return message(value=value, precision=precision, units=unit)
         return message(value=value, units=unit)
 
-    def resolve_unit(self, string_unit: str) -> Tuple[ord_schema.UnitMessage, int]:
+    def resolve_unit(self, string_unit: str) -> Tuple[ord_schema.UnitMessage, ord_schema.Message]:
```
Comment (qai222) [inline ord_schema/units.py:297]: I thought this was correct: the type check fail actually comes from `unit_synonyms` in `__init__` (the second key should be an enum value).

**Review by skearnes — APPROVED** [review]

File: ord_schema/units.py (hunk 4 above)
Comment (connorcoley) [reply ord_schema/units.py:297]: Could you clarify? All tests are passing but I will wait for you to resolve this comment (or help me figure out what's wrong) before merging

Comment (qai222) [reply ord_schema/units.py:297]: From this [line](https://github.com/open-reaction-database/ord-schema/blob/d23c1c3a7acf40b11ca8c8bca61ab8f625c4c67e/ord_schema/units.py#L220C9-L220C113) the second key in `unit_synonyms` is hinted to be an `ord_schema.Message` (causing the "wrong" return type in `resolve_unit`), but it should be just an ENUM option. So maybe it's better to change type hints there. It doesn't matter much tho.

**Pull request merged**